File: backend/app/services/razorpay_service.py
import razorpay
import os
from dotenv import load_dotenv

load_dotenv()

key_id = os.getenv("RAZORPAY_KEY_ID")
key_secret = os.getenv("RAZORPAY_KEY_SECRET")

client = None
if key_id and key_secret and not key_id.endswith("Here"):
    try:
        client = razorpay.Client(auth=(key_id, key_secret))
    except Exception as e:
        print(f"Failed to initialize Razorpay client: {e}")

def create_checkout_session(order_id, amount_in_rupees, customer_email, customer_name="", customer_phone=""):
    """
    Creates a Razorpay Payment Link for the given order.
    Returns a URL we can redirect the customer to (similar to Stripe Checkout).
    """
    try:
        if not client:
            print("RAZORPAY ERROR: Client not initialized. Check your RAZORPAY_KEY_ID and RAZORPAY_KEY_SECRET in .env.")
            return None

        if amount_in_rupees <= 0:
            print(f"RAZORPAY ERROR: Invalid amount ₹{amount_in_rupees}. Amount must be greater than zero.")
            return None

        amount_in_paise = int(amount_in_rupees * 100)
        callback_url = f"{os.getenv('FRONTEND_URL', 'http://localhost:5173')}/orders?success=true"

        payment_link = client.payment_link.create({
            "amount": amount_in_paise,
            "currency": "INR",
            "accept_partial": False,
            "description": f"Order #{str(order_id)}",
            "customer": {
                "email": customer_email,
                "name": customer_name,
                "contact": customer_phone
            },
            "notify": {
                "sms": False,
                "email": True
            },
            "reminder_enable": True,
            "callback_url": callback_url,
            "callback_method": "get"
        })
        
        return payment_link.get('short_url')

    except Exception as e:
        print(f"RAZORPAY ERROR: {e}")
        return None

File: backend_laravel/app/Services/RazorpayService.php
<?php

namespace App\Services;

use Razorpay\Api\Api;
use Exception;

class RazorpayService
{
    protected static $api;

    public static function init()
    {
        $keyId = env('RAZORPAY_KEY_ID');
        $keySecret = env('RAZORPAY_KEY_SECRET');

        if ($keyId && $keySecret && !str_contains($keyId, 'Here')) {
            try {
                self::$api = new Api($keyId, $keySecret);
            } catch (Exception $e) {
                \Log::error("Razorpay Init Error: " . $e->getMessage());
            }
        }
    }

    public static function createCheckoutSession($orderId, $amountInRupees, $customerEmail, $customerName = "", $customerPhone = "")
    {
        try {
            if (!self::$api) {
                self::init();
            }

            if (!self::$api) {
                return null;
            }

            if ($amountInRupees <= 0) {
                return null;
            }

            $amountInPaise = (int) ($amountInRupees * 100);
            $callbackUrl = env('FRONTEND_URL', 'http://localhost:5173') . '/orders?success=true';

            $paymentLink = self::$api->paymentLink->create([
                "amount" => $amountInPaise,
                "currency" => "INR",
                "accept_partial" => false,
                "description" => "Order #$orderId",
                "customer" => [
                    "email" => $customerEmail,
                    "name" => $customerName,
                    "contact" => $customerPhone
                ],
                "notify" => [
                    "sms" => false,
                    "email" => true
                ],
                "reminder_enable" => true,
                "callback_url" => $callbackUrl,
                "callback_method" => "get"
            ]);

            return $paymentLink['short_url'];
        } catch (Exception $e) {
            \Log::error("Razorpay Checkout Error: " . $e->getMessage());
            return null;
        }
    }
}
